include/aligned.h

```diff
@@ -14,21 +14,23 @@
    along with this program; if not, write to the Free Software
    Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1335  USA */
 
-#if defined __linux__
+#if defined HAVE_MALLOC_H
 # include <malloc.h>
 #endif
 
 inline void *aligned_malloc(size_t size, size_t alignment)
 {
 #ifdef _WIN32
   return _aligned_malloc(size, alignment);
-#elif defined __linux__
+#elif defined HAVE_MEMALIGN
   return memalign(alignment, size);
-#else
+#elif defined HAVE_POSIX_MEMALIGN
   void *result;
   if (posix_memalign(&result, alignment, size))
     result= NULL;
   return result;
+#else
+#error Unsupported aligned memory allocator
```

Comment:
I do not think that we really need any `HAVE_POSIX_MEMALIGN` check. We already have way too many compile-time feature checks, and those checks are executed by CMake in a single thread. Please do not make the build any slower.

We can simply assume that the `posix_memalign()` exists, just like we did so far. An error about missing `posix_memalign` should be as useful as any `#error` of our own.

Likewise, can we please avoid introducing `HAVE_MEMALIGN` on Linux?

Reply:
Fair point with `HAVE_POSIX_MEMALIGN`. The general idea is to support `memalign` in OSes other than Linux, so I think at least switching the `__linux__` to `HAVE_MEMALIGN` is worthwhile. Unless you are wanting a CMake directive that says "if not linux, then check for memalign?"

Reply:
On Linux and other platforms that we know about, we could do something similar as `cmake/os/WindowsCache.cmake` to avoid redundant compile-time checks:
```cmake
SET(HAVE_MEMALIGN 1 CACHE INTERNAL "")
```